### ISSUE TITLE
Catboost

### ISSUE THREAD

**Issue by pashashocky** [NixOS/nixpkgs#74395]
Would be a very useful package to have for ML Enthusiasts

**Project description**
Machine Learning Catboost Library

**Metadata**
CatBoost is a high-performance open source library for gradient boosting on decision trees

* homepage URL: https://catboost.ai/
* source URL: https://github.com/catboost/catboost
* license: Apache License 2.0 https://github.com/catboost/catboost/blob/master/LICENSE
* platforms: Linux
* installation instructions: https://catboost.ai/docs/installation/python-installation-method-build-from-source-linux-macos.html#python-installation-method-build-from-source-linux-macos

@jonringer Please let me know if I should stop tagging you in random issues and it bothers you (sorry, i'll stop), but I thought this might be of interest to you since you are very involved in tensorflow.

**Comment by pashashocky** [comment]
@teh @costrouc @abbradar taggining maintainers of xgboost and lightgbm. Wondering if you could look into compiling catboost too?

**Comment by jonringer** [comment]
I use tensorflow at work, which is why I care :). But I'm not a data-scientist

**Comment by pashashocky** [comment]
@jonringer understood, thank you. 

GPU support for catboost would also be very welcome.

**Comment by costrouc** [comment]
Looks like they have a non standard install script. But definitely doable. Sadly I don't think I have time in the near future but would be happy to point you in the right direction to get stared with a PR

**Comment by pashashocky** [comment]
> Looks like they have a non standard install script. But definitely doable. Sadly I don't think I have time in the near future but would be happy to point you in the right direction to get stared with a PR

 both eyes wide open - i'll try to start this w/e, any pointers very welcome, thank you.

**Comment by pashashocky** [comment]
Currently failing to build on nix: 

```
> ../../../ya make -r -DUSE_ARCADIA_PYTHON=no -DOS_SDK=local -DPYTHON_CONFIG=/nix/store/50j6awa5glyq1ssr9d2a8cqhym1lqlv3-user-packages/bin/python3.7-config
Downloading https://storage.mds.yandex.net/get-devtools-opensource/471749/ada0499bbcb40c08b739da8259d73011 [..] OK
Downloading https://storage.mds.yandex.net/get-devtools-opensource/471749/1227700075 [.....................................................................] OK
ERROR: [Errno 2] No such file or directory: '/home/pash/.ya/tools/caca278aee_d/ya-bin'
```

**Comment by teh** [comment]
@pashashocky - maybe it's easier if you create a work-in-progress pull request so other people can check? Two things that stand out are:

1. It looks like something is being downloaded during the build process. This won't work with sandboxing because there won't be a working network.
2. Nix builds should be isolated, i.e. access to `/home` should be rewritten to the local build directory, not `/home` (probably hardcoded somewhere in the build scripts).

Packaging for nix can be quite fiddly - I think [arrow-cpp](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/arrow-cpp/default.nix) might be a good example for fetching complex dependencies + build script patches.
